components/datetime/src/provider/calendar/mod.rs

```diff
@@ -31,7 +31,17 @@ pub use symbols::*;
     marker(IndianDateLengthsV1Marker, "datetime/indian/datelengths@1"),
     marker(EthiopianDateLengthsV1Marker, "datetime/ethiopic/datelengths@1"),
     marker(RocDateLengthsV1Marker, "datetime/roc/datelengths@1"),
-    marker(PersianDateLengthsV1Marker, "datetime/persian/datelengths@1")
+    marker(PersianDateLengthsV1Marker, "datetime/persian/datelengths@1"),
+    marker(
+        IslamicObservationalDateLengthsV1Marker,
+        "datetime/islamicobservational/datelengths@1"
+    ),
+    marker(IslamicCivilDateLengthsV1Marker, "datetime/islamiccivil/datelengths@1"),
+    marker(UmmAlQuraDateLengthsV1Marker, "datetime/ummalqura/datelengths@1"),
```

Comment:
nit: we should call these codes  `islamic_observational`, `islamic_civil`, `islamic_ummalqura`, `islamic_tabular`, etc

Reply:
I'd prefer something shorter, like we collapsed `japanese_extended` into `japanext`. `islobv`, `islciv`, `isluaq`, `isltab`.  We can actually also use slashes, `islamic/obv`, `islamic/civ`, `islamic/uaq`, `islamic/tab` would also be a nice structure.